server/client_test.go

```diff
@@ -852,6 +852,35 @@ func TestSplitSubjectQueue(t *testing.T) {
 	}
 }
 
+func TestTypeString(t *testing.T) {
+	_, c, _ := setupClient()
+
+	cases := []struct {
+		intType    int
+		stringType string
+	}{
+		{intType: CLIENT, stringType: "Client"},
+		{intType: ROUTER, stringType: "Router"},
+		{intType: GATEWAY, stringType: "Gateway"},
+		{intType: LEAF, stringType: "Leafnode"},
+		{intType: JETSTREAM, stringType: "JetStream"},
+		{intType: ACCOUNT, stringType: "Account"},
+		{intType: SYSTEM, stringType: "System"},
+		{intType: -1, stringType: "Unknown Type"},
+	}
+	for _, cs := range cases {
+		_, c, _ := setupClient()
+		c.kind = cs.intType
```

Comment:
I ran this test and it would leave client's go routine running. Doing a defer c.Close() after setup would cause some failures because in certain case it would try to reconnect, etc.. We don't need the client to be setup for this test, so simple replace with:
```suggestion
		c := &client{kind: cs.intType}
```

Reply:
fixed